app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/state/MessageComposerState.kt

```diff
@@ -23,400 +23,111 @@ package com.wire.android.ui.home.messagecomposer.state
-import androidx.compose.ui.text.TextRange
-import androidx.compose.ui.text.input.TextFieldValue
-import com.wire.android.appLogger
 import com.wire.android.ui.home.conversations.model.UIMessage
-import com.wire.android.ui.home.conversations.model.UIMessageContent
-import com.wire.android.ui.home.conversations.model.UIQuotedMessage
-import com.wire.android.ui.home.conversations.selfdeletion.SelfDeletionMapper.toSelfDeletionDuration
-import com.wire.android.ui.home.messagecomposer.model.UiMention
-import com.wire.android.ui.home.newconversation.model.Contact
 import com.wire.android.ui.theme.wireColorScheme
-import com.wire.android.util.EMPTY
-import com.wire.android.util.MENTION_SYMBOL
-import com.wire.android.util.NEW_LINE_SYMBOL
-import com.wire.android.util.WHITE_SPACE
-import com.wire.android.util.ui.toUIText
 import com.wire.kalium.logic.data.message.mention.MessageMention
-import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.feature.conversation.InteractionAvailability
+import com.wire.kalium.logic.feature.conversation.SecurityClassificationType
 import com.wire.kalium.logic.feature.selfDeletingMessages.SelfDeletionTimer
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.StateFlow
-import kotlin.time.Duration.Companion.ZERO
+import kotlin.time.Duration
 
+@Suppress("LongParameterList")
 @Composable
-fun rememberMessageComposerState(): MessageComposerState {
+fun rememberMessageComposerState(
+    isFileSharingEnabled: Boolean = true,
+    selfDeletionTimer: SelfDeletionTimer = SelfDeletionTimer.Enabled(Duration.ZERO),
+    interactionAvailability: InteractionAvailability = InteractionAvailability.ENABLED,
+    securityClassificationType: SecurityClassificationType = SecurityClassificationType.NONE,
+    onShowEphemeralOptionsMenu: () -> Unit,
+    searchMentions: (String) -> Unit,
+    onSendMessage: (MessageBundle) -> Unit
+): MessageComposerState {
     val context = LocalContext.current
 
-    val mentionSpanStyle = SpanStyle(
+    val mentionStyle = SpanStyle(
         color = MaterialTheme.wireColorScheme.onPrimaryVariant,
         background = MaterialTheme.wireColorScheme.primaryVariant
     )
 
-    val focusManager = LocalFocusManager.current
-    val inputFocusRequester = FocusRequester()
-
     return remember {
         MessageComposerState(
             context = context,
-            focusManager = focusManager,
-            inputFocusRequester = inputFocusRequester,
-            mentionSpanStyle = mentionSpanStyle
+            isFileSharingEnabled = isFileSharingEnabled,
+            interactionAvailability = interactionAvailability,
+            securityClassificationType = securityClassificationType,
+            onShowEphemeralOptionsMenu = onShowEphemeralOptionsMenu,
+            selfDeletionTimer = selfDeletionTimer,
+            mentionStyle = mentionStyle,
+            searchMentions = searchMentions,
+            onSendMessage = onSendMessage
```

Comment:
All these values are only set once and will never change in the `MessageComposerState` because of the `remember` - is it intentional that these are only initial values? If so then maybe they should have names indicating that. If not, then I'm afraid it won't work as expected. 
I can see that for instance `selfDeletionTimer` parameter is used inside `MessageComposerState` only to set initial values, but `securityClassificationType` is used in `MessageComposerInput` as a state, but since it's in `remember` without any keys, it will never be changed so it may be a problem. 
I still don't quite understand why we create states that are responsible for some actions. For me state should be a data class whose only purpose is to keep the current state and not execute any actions.

Reply:
thanks for the feedback 🙇 I do not understand why you think it will never be changed ? It can still be changed if it is a mutable state object inside the MessageComposerState, MessageComposerState is only a state "holder" meaning it will hold all the values during the recomposiition as it will not be recreated includiing the mutable state objects, but that does not mean the mutable state object cannot change the values. 

Also what do you mean with 
I still don't quite understand why we create states that are responsible for some actions. For me state should be a data class whose only purpose is to keep the current state and not execute any actions.

could u pin-point me maybe with a code what the problem is and what do you refer to as an action ?

Reply:
if you refer to , searchmentions and onSendMessage and onShowEphemneralOptionsMenu, this is made so that different sealed class instance can control stuff from the outside world, for example : 

When we are displaying self deleting input, we only then can have the action to show ephemeral options menu, so we need to hold the reference to the lambda passed during remember phase inside ConversationScreen, however now u are saying this, perhaps you are right, this is something we never did or do anywhere so maybe this will only more confusion. I'll try to look for an alternative, if you have some better idea how to make it work, pls let me know here. 

I was thinking about exposing a lambda from within MessageComposer Composable, I think that is what most Composable from material3 are doing too.

Reply:
It will never change when `MessageComposerViewState` changes because it's wrapped in `remember` without any key, which means once the value is stored, it won't be recalculated, so `MessageComposerState` will be created with the values that are passed to `rememberMessageComposerState` the first time it's called and even if one of the states in `MessageComposerViewState` changes, `MessageComposerState` will still hold the first values. In practise, for instance, if `securityClassificationType` is being updated in `messageComposerViewState.securityClassificationType` then this change won't be reflected in `MessageComposerState` because it remembered previous value, so it will keep using previous value in `EnabledMessageComposer`.

About state that contains some actions, yep - I was thinking about `sendMessage` action inside `MessageComposerState`. I just feel like state classes should just hold the current values and it's the ViewModel which should make actions and update/change these states. Also, I can see that for you it's perfectly clear but I'm not quite sure I understand the difference between "state" and "stateholder" in that scenario 😅

Reply:
as agreed per call I am going to adjust it a bit

Reply:
looks better now 👌